### DISCUSSION TITLE
Extend Account Console v3

### DISCUSSION THREAD

**Discussion by MariiaAlekseichuk** [keycloak/keycloak#25533] [Q&A] [answered]
Hi all.

I've found a quickstart about how to extend Account Console v2. But as I see it will be deprecated and v3 is going to be actual.
But there are no any information how to extend Account Console v3.

Could you please provide any?
Did anyone tried and got succeeded?

**Comment by ssilvert** [comment, marked as answer]
Much of what you see in the Account Console V2 quickstart will be applicable to V3.  For instance, we will still have the content.json file that you can use to add and remove sections.

There are also enhancements such as that if you are using the new User Profile feature, then the Personal Info section will be built dynamically based on your User Profile configuration.

We plan to have a new quickstart and more documentation for all of the available customization features when V3 is fully released.

**Reply by MariiaAlekseichuk** [reply]
Thanks a lot for this information!
Do you have any planned date when V3 should be fully released?

**Reply by Fireblade954** [reply]
to hook into this discussion, i there a way to not render the "applications" tab in the menu? i already created a empty content.json but the default 3 menu items are always rendered. I see something is based on settings but as far as i can see there isnt a setting that hides the applications tab, or even the "personal info" tab. We have a situation where we have disabled all user profile fields using the new v3 abilities. So now the "personal info" tab is empty and useless we would like to hide that.

**Comment by ssilvert** [comment]
> Thanks a lot for this information! Do you have any planned date when V3 should be fully released?

We are shooting for Keycloak 24.

**Comment by ssilvert** [comment]
> to hook into this discussion, i there a way to not render the "applications" tab in the menu? i already created a empty content.json but the default 3 menu items are always rendered. I see something is based on settings but as far as i can see there isnt a setting that hides the applications tab, or even the "personal info" tab. We have a situation where we have disabled all user profile fields using the new v3 abilities. So now the "personal info" tab is empty and useless we would like to hide that.

I believe the content.json stuff should be working the same way as in V2 where you can set a "hidden" flag to hide a section.  But content.json functionality was only added recently.  So you might need to check out the nightly build.  I'm not sure if it's in the latest patch release.

@edewit Can you verify what I said above?

**Reply by edewit** [reply]
We didn't want to re-introduce this `hidden` property as it was implemented in v2, because it was using `eval`! That's why initially we made the menu static, but I've created [this PR](https://github.com/keycloak/keycloak/pull/25744) to re-introduce that behaviour, with `hidden` only controllable over the features

**Comment by pstrobel** [comment]
Hi Stan, Erik
I've just tried out the extend-account-console quickstart with v24, but it doesn't render the menu and the description seems to be out of sync with the code. Is there an ETA for this? Or, if not -  could give me some pointers how it's meant to work now so I can update the quickstart?

**Comment by Sieboldianus** [comment]
I upgraded to Keycloak 25.0.0 today and unfortunately, my modifications to the Account UI are not visible anymore. I just want to change the color, nothing more. I cannot find the artifacts for the new Vue Account UI to modify this. Can someone point me to where I can find this?

**Reply by Sieboldianus** [reply]
I found out: The CSS definitions simply changed. See my modifications of V3 account theme [here](https://framagit.org/lausitzer-surfer-eV/mitglieder-management/account-kc-theme)

![image](https://github.com/keycloak/keycloak/assets/13646666/3ff4a926-6f60-409f-97d3-29be707dc3f4)
